quic/s2n-quic-core/src/buffer/duplex/split.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    buffer::{
+        duplex,
+        reader::{self, storage::Infallible as _, Reader, Storage as _},
+        writer::{self, Writer},
+        Error,
+    },
+    varint::VarInt,
+};
+use core::convert::Infallible;
+
+pub struct Split<'a, C, D>
+where
+    C: writer::Storage + ?Sized,
+    D: duplex::Skip<Error = Infallible> + ?Sized,
+{
+    chunk: &'a mut C,
```

Comment:
This struct is intended to be used for when you get a packet from the network and need to put it somewhere (think `AsyncRead`). The application provides a buffer to write data into. We also have a reassembler (the `duplex`) in case the application buffer isn't big enough or the data arrives out of order.

So to answer your question: the `chunk` here is being provided temporarily by the application.